src/System.Text.Encoding.CodePages/src/System/Text/SBCSCodePageEncoding.cs

```diff
@@ -852,6 +852,7 @@ public override unsafe int GetCharCount(byte* bytes, int count, DecoderNLS decod
             byte[] byteBuffer = new byte[1];
             char* charEnd = chars + charCount;
 
+            Debug.Assert(decoder != null);
             DecoderFallbackBufferHelper fallbackHelper = new DecoderFallbackBufferHelper(decoder.FallbackBuffer);
```

Comment:
I know we didn't hit this problem before but the right fix is to change this line to 

DecoderFallbackBufferHelper fallbackHelper = new DecoderFallbackBufferHelper(decoder ==null ? DecoderFallback.CreateFallbackBuffer() : decoder.FallbackBuffer);

and please revert your changes in this file.

thanks for catching that.